Document Query DSL and Highlight #334 (#335)
enonic/app-guillotine: docs/api.adoc

Boolean DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|should: [<<QueryDSLInput>>]
|The query should appear in the matching document.

|must: [<<QueryDSLInput>>]
|The query must appear in matching documents and will contribute to the score.

|mustNot: [<<QueryDSLInput>>]
|The query must not appear in the matching documents

|filter: [<<QueryDSLInput>>]
|The query must appear in matching documents.

|boost: Float
|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.

|===

=== NgramDSLExpressionInput

Ngram DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|fields: [String]!
|List of fields (propertyPaths) to include in the search.

|query: String!
|A query string to match field value(s). Support the set of operators.

|operator: [<<DslOperatorType>>]
|DSL operator. By default, `OR` (any of the words in the query matches).

|===

=== StemmedDSLExpressionInput

Stemmed DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|fields: [String]!
|List of fields (propertyPaths) to include in the search.

|query: String!

> |A query string to match field value(s). Support the set of operators.

Reviewer: ```suggestion
|A query string to match field value(s). Supports the set of https://developer.enonic.com/docs/xp/next/storage/dsl#search_string_operators[operators].
```

> |language: String!
> |A query string to match field value(s). Support the set of operators.

Reviewer: ```suggestion
|language: String!
|Content language that was used for stemming. List of https://developer.enonic.com/docs/xp/next/storage/indexing#languages[supported languages]
```

|operator: <<DslOperatorType>>
|DSL operator. By default, `OR` (any of the words in the query matches).

|boost: Float
|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.

|===

=== FulltextDSLExpressionInput

Fulltext DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|fields: [String]!
|List of fields (propertyPaths) to include in the search.

|query: String!

> |A query string to match field value(s). Support the set of operators.

Reviewer: |query: String!
|A query string to match field value(s). Supports the set of https://developer.enonic.com/docs/xp/next/storage/dsl#search_string_operators[operators].

|Property name to search.

|boost: Float
|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.

|stringValues: [String]
|String values.

|doubleValues: [Float]
|Float values.

|longValues: [Int]
|Integer values.

|booleanValues: [Boolean]
|Boolean values.

|localDateValues: [Date]
|Date values. Element of array is a date representation. Will be indexed with UTC timezone offset. For instance, `2015-03-16`.

|localDateTimeValues: [LocalDateTime]
|LocalDateTime values. Element of array is a date-time representation without timezone. Will be indexed with UTC timezone offset. For instance, `2015-03-16T10:00:02`.

|localTimeValues: [LocalTime]
|LocalTime values. Element of array is a time representation without date or timezone(nor subsecond). For instance, `10:00:03`.

|instantValues: [DateTime]
|DateTime values. Element of array is a single point on the time-line (may include subsecond up to 9 digits). For instance, `2015-03-16T10:00:02Z`.

|===

=== ExistsDSLExpressionInput

Exists DSL expression input type. Returns nodes that contain a value for a field.

==== Fields

|===
|Field name: Type | Description

|field: String!

> |Name of a field to check on existence.

Reviewer: ```suggestion
|Name of a field to check for existence.
```

|boost: Float
|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.

|===

=== GeoPointSortDslInput

GeoPoint sort DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|lat: Float!
|Latitude.

|lon: Float!
|Longitude.

|===

=== SortDslInput

Sort DSL expression input type.

==== Fields

|===
|Field name: Type | Description

|field: String!

> |Name of a field to check on existence.

Reviewer: ```suggestion
|Name of a field to sort by.
```
